### PR TITLE
Drop Mock dependency, use FakeTransaction, FakeSystem and Transaction.to_map/1 instead

### PR DESCRIPTION
**Pull request by jeffkreeftmeijer**
We support Elixir 1.8, but some tests are failing caused by the Mock dependency. Since we planned on removing that already, I went ahead and took it out in favor of our Fake- mocks (noun!).

The Fake Appsignal modules were moved to their own subdirectory, giving us a `test/support/fake_system.ex` and a `test/support/appsignal/fake_system.ex`, which are fakes for `System` and `Appsignal.System`, respectively.

### CODE REVIEW

File: test/support/appsignal/fake_system.ec [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  def root?, do: get(__MODULE__, :root)
+
+  def heroku?, do: get(__MODULE__, :heroku)
+
+  def uid, do: get(__MODULE__, :uid)
+
+  def priv_dir, do: get(__MODULE__, :priv_dir)
+
+  def agent_platform, do: get(__MODULE__, :agent_platform)
+end
```
Comment (tombruijn) [inline test/support/appsignal/fake_system.ec]: do we need this file?

Comment (jeffkreeftmeijer) [reply test/support/appsignal/fake_system.ec]: Rename-fail. Fixed.

**Comment by tombruijn** [comment]
@jvanbaarsen do you have time to look at this? otherwise I'll merge it

**Comment by jvanbaarsen** [comment]
@tombruijn I went over it, but wanted to ping @jeffkreeftmeijer to see if he has time to go over it with me someday. Because now I'm looking at code that looks "ok" but doesn't say much to me

**Pull request merged**